src/index.ts

```diff
@@ -60,10 +71,47 @@ async function gameIterator(path) {
       metric.processGame(Array.from(historyGenerator), metadata);
     }
   }
-  pieceLevelMoveInfoMetric.aggregate();
-  pieceLevelMoveInfoMetric.logResults();
+  results['Number of games analyzed'] = gameCounter;
+  let metricCallsCount = 0;
+  for (const metric of metrics) {
+    metricCallsCount++;
+    results[metric.constructor.name] = metric.aggregate()
+  }
 }
```

Comment:
I'm not sure your boardMap/visualization work merges well with the index.ts file i had set up previously to work with the metrics. It might be worth breaking out the boardMap stuff into a separate index.ts file (maybe "indexViz.ts")

Reply:
because the current index.ts prints either to console or results.json, which you might not necessarily want

Reply:
hmm I'm not sure what you're talking about here

Reply:
don't think I ever really got that piece working

Reply:
I think we'd be better off writing the viz from scratch, it's pretty trivial tbh. heap map is just a color scale on an 8x8 grid

Reply:
removing viz stuff for now then